backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/DiagramComponentProps.java

```diff
@@ -32,10 +34,13 @@
 
     private final Optional<Diagram> previousDiagram;
 
-    public DiagramComponentProps(VariableManager variableManager, DiagramDescription diagramDescription, Optional<Diagram> previousDiagram) {
+    private final List<ViewCreationRequest> viewCreationRequests;
+
+    public DiagramComponentProps(VariableManager variableManager, DiagramDescription diagramDescription, List<ViewCreationRequest> viewCreationRequests, Optional<Diagram> previousDiagram) {
         this.variableManager = Objects.requireNonNull(variableManager);
         this.diagramDescription = Objects.requireNonNull(diagramDescription);
         this.previousDiagram = Objects.requireNonNull(previousDiagram);
+        this.viewCreationRequests = Objects.requireNonNull(viewCreationRequests);
```

Comment:
Should make an immutable copy instead of keeping a reference to a potentially mutable `List `.